src/main/java/org/wildfly/security/auth/client/AuthenticationConfiguration.java

```diff
@@ -1172,17 +1161,25 @@ public AuthenticationConfiguration useMechanismProperties(Map<String, ?> mechani
     }
 
     /**
-     * Create a new configuration which is the same as this ocnfiguration, but which uses the given kerberos security
+     * Create a new configuration which is the same as this configuration, but which uses the given kerberos security
      * factory to acquire the GSS credential required for authentication.
      *
      * @param kerberosSecurityFactory a reference to the kerberos security factory to be use
      * @return the new configuration
+     * @deprecated to specify kerberos credential use {@link #useCredential(Credential)} or {@link #useGSSCredential(GSSCredential)} method instead
      */
+    @Deprecated
     public AuthenticationConfiguration useKerberosSecurityFactory(SecurityFactory<Credential> kerberosSecurityFactory) {
-        if (Objects.equals(this.kerberosSecurityFactory, kerberosSecurityFactory)) {
-            return this;
+        if (kerberosSecurityFactory != null) {
```

Comment:
Need to note this changes behavior of deprecated method when calling to unset kerberosSecurityFactory:
```
ac.useKerberosSecurityFactory(null)
```
* in previous code it will ensure principal from kerberos factory will not be used
* in new code it will keep already obtained credential in credentials to use

On the other hand, behavior for calling with null kerberosSecurityFactory is not specified in javadoc - I think we can ignore this...

Reply:
I just realized that the behavior is completely different, because the original version created the credential from the factory at the time of use, but I am creating it right away. Would creating new CredentialSource class that uses SecurityFactory (same way as KeyStoreCredentialSource does) be a good idea? Then we could easily remove the CredentialCource from configuration.